File: src/RentCars.Domain/Vehicles/NameOfVehicle.cs
namespace RentCars.Domain.Vehicles;

public class NameOfVehicle
{
    public Guid Id { get; }
    public String Brand { get; }
    public String Model { get; }

    public NameOfVehicle(Guid id, String brand, String model)
    {
        Id = id;
        Brand = brand;
        Model = model;
    }
}

File: API/api.client/src/domain/vehicles/nameOfVehicle.ts
export class NameOfVehicle {
    constructor(
        public id: string,
        public brand: string,
        public model: string
    ) { }
}

export function mapToNameOfVehicle(data: any) {
    return new NameOfVehicle(
        data.id,
        data.brand,
        data.model
    );
}